File: app/src/main/java/com/soonyong/mywebcrawler/config/manage/ConfigManager.java
package com.soonyong.mywebcrawler.config.manage;

import com.soonyong.mywebcrawler.config.CrawlConfig;

import java.io.IOException;

public interface ConfigManager {
    CrawlConfig getCrawlConfig() throws IOException;

    void addCrawlConfigTarget(CrawlConfig.Target target) throws IOException;
}

File: app/src/main/java/com/soonyong/mywebcrawler/config/manage/ConfigManager.kt
package com.soonyong.mywebcrawler.config.manage

import com.soonyong.mywebcrawler.config.CrawlConfig
import java.io.IOException

interface ConfigManager {
    val crawlConfig: CrawlConfig

    @Throws(IOException::class)
    fun addCrawlConfigTarget(target: CrawlConfig.Target)
}